utils/python/CIME/XML/generic_xml.py

```diff
@@ -63,55 +58,72 @@ def write(self, outfile=None):
             outfile = self.filename
 
         logger.debug("write: " + outfile)
-        xmlstr = ET.tostring(self.root)
-        doc = minidom.parseString(xmlstr)
-        with open(outfile,'w') as xmlout:
-            doc.writexml(xmlout,addindent='  ')
+        try:
```

Comment:
If the xml cannot be translated to a string then dump self.root - this usually points directly to the offending entry.